pkg/upgraders/healthcheck_pdb.go

```diff
@@ -0,0 +1,84 @@
+package upgraders
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/managed-upgrade-operator/pkg/dvo"
+	"github.com/openshift/managed-upgrade-operator/pkg/metrics"
+	policyv1 "k8s.io/api/policy/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// HealthCheckPDB performs a health check on the PodDisruptionBudget (PDB) metrics.
+// It returns true if the health check passes, false otherwise.
+// It also returns an error if there was an issue performing the health check.
+func HealthCheckPDB(metricsClient metrics.Metrics, c client.Client, dvo dvo.DvoClientBuilder) (bool, error) {
+
+	err := checkPodDisruptionBudgets(c)
+	if err != nil {
+		return false, err
+	}
+
+	err = checkDvoMetrics(c, dvo)
+	if err != nil {
+		return false, err
+	}
+
+	// Health check passed
+	return true, nil
+}
+
+func checkPodDisruptionBudgets(c client.Client) error {
+	// List all PodDisruptionBudgets
+	pdbList := &policyv1.PodDisruptionBudgetList{}
+	err := c.List(context.TODO(), pdbList)
+	if err != nil {
+		fmt.Print("unable to list PodDisruptionBudgets/v1")
+		return err
+	}
+
+	for _, pdb := range pdbList.Items {
+		if !strings.HasPrefix(pdb.Namespace, "openshift-*") {
+			if pdb.Spec.MaxUnavailable != nil && pdb.Spec.MaxUnavailable.IntVal == 0 {
+				//BAD pdb
+				return fmt.Errorf("found a PodDisruptionBudget with MaxUnavailable set to 0")
+			} else if pdb.Status.CurrentHealthy < pdb.Status.DesiredHealthy {
+				//BAD pdb
+				return fmt.Errorf("found a PodDisruptionBudget with CurrentHealthy less than DesiredHealthy")
+			}
+		}
+	}
```

Comment:
Some points to note as discussed:
-  Based on https://github.com/openshift/managed-cluster-config/blob/master/deploy/managed-upgrade-operator-config/10-managed-upgrade-operator-configmap.yaml#L78 we do have few `openshift-*` namespace that we want to treat as user namespaces which might have workloads installed by users and are not platform workloads. We can include them as an exception to `openshift-*` pattern.
- The `else` block could have more implications since it could refer to a transient state which will resolve in some time (or may not resolve in some time). We can start with the static checks in the `if` block only. 
- It would also help to include returning the `reason` label from this function if we are considering different scenarios for PDB failure.

Reply:
Updated changes,Thank you.